network/wifi/at_parser.c

```diff
@@ -204,12 +184,41 @@ struct at_desc {
 	}			ipd_stat;
 	/* Variable where current message length is built */
 	uint32_t		ipd_len;
+	/* Will be called when a new connection is created or closed */
+	void			(*connection_callback)(void *ctx, enum at_event,
+			uint32_t conn_id, struct circular_buffer **cb);
+	/* Context that will be passed to the callback */
+	void			*callback_ctx;
 };
 
 /******************************************************************************/
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 
+static void dispatch_event(struct at_desc *desc, enum at_event event)
+{
+	struct connection_desc *conn;
+
+	conn = &desc->conn[desc->current_conn];
+
+	if (event == AT_NEW_CONNECTION) {
```

Comment:
if this function will be extended to support more events it would make sense to shift to a switch.

Reply:
Just this 2 events will be used